internal/idp/providers/ldap/session.go

```diff
@@ -14,45 +15,118 @@ import (
 )
 
 var ErrNoSingleUser = errors.New("user does not exist or too many entries returned")
+var ErrFailedLogin = errors.New("user failed to login")
 
 var _ idp.Session = (*Session)(nil)
 
 type Session struct {
 	Provider *Provider
 	loginUrl string
-	user     string
-	password string
+	User     string
+	Password string
 }
 
 func (s *Session) GetAuthURL() string {
 	return s.loginUrl
 }
 func (s *Session) FetchUser(_ context.Context) (idp.User, error) {
-	l, err := ldap.DialURL("ldap://" + s.Provider.host + ":" + s.Provider.port)
+	var user *ldap.Entry
+	for _, server := range s.Provider.servers {
+		userT, err := tryLogin(
+			server,
+			s.Provider.startTLS,
+			s.Provider.bindDN,
+			s.Provider.bindPassword,
+			s.Provider.baseDN,
+			s.Provider.getNecessaryAttributes(),
+			s.Provider.userObjectClasses,
+			s.Provider.userFilters,
+			s.User,
+			s.Password,
+			s.Provider.timeout,
+		)
+		if err != nil {
```

Comment:
if multiple servers are configured, but the first server is not reachable, the user is not able to login

Reply:
generally the `try` should only handle connection errors to solve HA scenarios, right?